### PR TITLE
New mongo find update setup

### PR DESCRIPTION
**Pull request by joncodo**
- Have only one db connection per collection task
- Fix the jira issues collection method
- Refactor methods
- Async bulk with Promise.all

### CODE REVIEW

File: db/mongo/connection.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ module.exports = {
   },
   disconnect: (client) => {
     try {
+			console.log('INFO >>> closing mongo connection')
```
Comment (paulgoertzen-merico) [inline db/mongo/connection.js:19]: Weird tabbing bro

File: src/collection/collectors/jira/changelogs.js [hunk 2]
```diff
@@ -6,29 +6,25 @@ const fetcher = require('./fetcher')
...
+          issueId: issue.id,
+          ...change
+        }))
       }
-    } catch (error) {
-      console.error(error)
-    } finally {
-      dbConnector.disconnect(client)
     }
+    await Promise.all(promises)
```
Comment (paulgoertzen-merico) [inline src/collection/collectors/jira/changelogs.js:27]: Now there's no try catch, and everything will stop if one changelog fails to insert. Is that what you want?

File: src/enrichment/enrichers/jira.js [hunk 3]
```diff
@@ -53,8 +93,8 @@ module.exports = {
       }
     }
 
-    return isDone
-      ? Math.round(leadTime / 1000)
-      : 0
+    return isDone ?
+      Math.round(leadTime / 1000) :
+      0
```
Comment (paulgoertzen-merico) [inline src/enrichment/enrichers/jira.js:98]: Why change this formatting? Pretty sure the way it was is the standard way

File: src/enrichment/enrichers/jira.js [hunk 4]
```diff
@@ -3,34 +3,74 @@ require('module-alias/register')
...
+    issues.forEach(async issue => {
+      leadTimePromises.push(module.exports.calculateLeadTime(issue, options.db))
+      issuesToCreate.push({
         id: issue.id,
         url: issue.self,
         title: issue.fields.summary,
         projectId: issue.fields.project.id,
-        description: issue.fields.description,
-        leadTime
+        description: issue.fields.description
```
Comment (paulgoertzen-merico) [inline src/enrichment/enrichers/jira.js:50]: Can you comment out description? Hezheng and I found out yesterday that description is just a crazy object with all kinds of nested stuff. It's gonna take a while to figure out how to work with it.

**Review by paulgoertzen-merico — COMMENTED** [review]
Overall it's good. I like that the db is being passed as an argument (which we'll need to do to use the plugin pattern). I'm not totally sure the use of Promise.all() is the right choice though, as it will stop all insertion as soon as one fails. I feel like it'd be better to allow the occasional failure for now.

File: src/enrichment/enrichers/jira.js (hunk 3 above)
Comment (joncodo) [reply src/enrichment/enrichers/jira.js:98]: Must have been my prettier format. I did not intentionally do this

File: src/enrichment/enrichers/jira.js (hunk 4 above)
Comment (joncodo) [reply src/enrichment/enrichers/jira.js:50]: ok

File: src/collection/collectors/jira/changelogs.js (hunk 2 above)
Comment (joncodo) [reply src/collection/collectors/jira/changelogs.js:27]: that is true. I need to think about this one

**Pull request merged**